release-notes.txt

```diff
@@ -1,3 +1,622 @@
+ - Support for IN865 region in GNRC LoRaWAN
+ - 83 additional modules supported by Kconfig
+ - 14 new boards, 6 new device drivers, 8 packages updated
+
+[UPDATE!!!]
+532 pull requests, composed of 1439 commits, have been merged since the
+last release, and 65 issues have been solved. XXX people contributed with
+code in 114 days. 3852 files have been touched with 348615 (+) insertions and
+52513 deletions (-).
+
+Notations used below:
+=====================
+
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+New features and changes
+========================
+
+Core
+----
++ msg_bus: allow to differ between messages form bus and from thread
++ mutex: Add mutex_cancel
++ mbox: add mbox_size(), mbox_avail()
+* mutex: compatibility with non-C languages
+* refactor thread info 
+- remove kernel_types.h
+
+System Libraries
+----------------
++ Add thread-safe Malloc
++ sys/tsrb: Make thread-safe
++ mtd_mapper: implement pagewise functions
++ xtimer: introduce xtimer_msleep()
++ usbus: Add USB peripheral serial string support
++ hashes: Add PBKDF2-sha256 implementation
++ usbus/dfu: add Device Firmware Upgrade support for USBUS
++ Add helpers for pointer tagging
++ atomic_utils: Functions for atomic access
++ entropy_source: add new module for entropy sources
++ nimble_autoadv: add new module
+* ztimer:
+    + add ztimer_mutex_lock_timeout()
+    * fix xtimer_compat
+    * increase xtimer_compat coverage
+* oneway-malloc: check calloc args
+* event: Allow single event_thread handling all queues
+- irq_handler: Drop module
+
+Networking
+----------
++ gnrc_netif: add capability to join or leave link layer multicast groups
++ gnrc_sixlowpan_frag:
+    + initial import of minimal forwarding
+    + initial import of Selective Fragment Recovery
++ gnrc_netif_ethernet: Support RX timestamp
++ Initial implementation of IEEE 802.15.4 security
+* sys/net/gnrc: fix in-band signaling of no RSSI
+* gnrc_netif_pkq: fix pktbuf leak
+* Fixed a critical bug, allowing to crash the whole system with a specially
+  crafted LoRa frame
+* GNRC LoRaWAN:
+    + Support for IN865 region, Improve debugging
+    * GNRC LoRaWAN: Bug fix for zero CFList
+* IEEE 802154 Radio HAL:
+    + add `read` function
+    + add IEEE802154_RADIO_INDICATION_CRC_ERROR
+    + add IEEE802154_RADIO_INDICATION_TX_START
+    + add helpers for RSSI and dBm conversion
+    * fix wrong rssi conversions treewide
+* IEEE 802.15.4 SubMAC:
+    + add retransmission reporting
+    * enable ACK_REQ by default
+    * unify legacy and radio HAL bootstrap
+* Gcoap: 
+    + Process CON responses
+    * make blocksize configurable
+- gcoap: remove deprecated gcoap_add_qstring
+- gnrc_nettype: remove GNRC_NETTYPE_IOVEC
+- gnrc_pktbuf: drop gnrc_pktbuf_replace_snip() 
+- gnrc_sixlowpan_rb: remove GNRC_SIXLOWPAN_FRAG_RBUF_AGGRESSIVE_OVERRIDE
+- sock/dtls: remove sock_dtls_session_create due to deprecation
```

Comment:
(I fixed all trailing spaces in a follow up commit ;)

Reply:
> is [8f2912f](https://github.com/RIOT-OS/RIOT/commit/8f2912fc2559f969fb8758e5ae1006cc1da2370a) ok?

8f2912fc2559f969fb8758e5ae1006cc1da2370a does not remove the duplicate remove notes here, if I am not mistaken.

Reply:
Yepp, [still there](https://github.com/RIOT-OS/RIOT/pull/15913/files#diff-f41b6887f65b706736aec3e6716bce4844a2aa169cba5e675c630f10b4fdbcecR116-R120) and still not summarized with the point `- Remove deprecated features`. In case my comment was unclear, IFF you want to keep the duplicates here, please apply the following suggestion:

```suggestion
- Remove deprecated features
  - gcoap: remove deprecated gcoap_add_qstring
  - gnrc_nettype: remove GNRC_NETTYPE_IOVEC
  - gnrc_pktbuf: drop gnrc_pktbuf_replace_snip() 
  - gnrc_sixlowpan_rb: remove GNRC_SIXLOWPAN_FRAG_RBUF_AGGRESSIVE_OVERRIDE
  - sock/dtls: remove sock_dtls_session_create due to deprecation
```

Reply:
Gnarf, due to this being outdated, the suggestion can't be commited. I add another comment to the new position.

Reply:
>  add another comment to the new position.

https://github.com/RIOT-OS/RIOT/pull/15913/files#r568871192

Reply:
What I did there was only group them in the Deprecated section. Sure, I can also group them here.
I think it won't hurt to have them duplicated, since they reflect on one side changes in the codebase and on the other side deprecated modules.

Reply:
done! thanks

Reply:
can I mark it as resolved?

Reply:
Yepp